eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -148,12 +147,14 @@ object Commitments {
     // the HTLC we are about to create is outgoing, but from their point of view it is incoming
     val outgoingHtlcs = reduced.htlcs.filter(_.direction == IN)
 
-    // a node cannot spend pending incoming htlcs, and need to keep funds above the reserve required by the counterparty, after paying the fee
-    // we look from remote's point of view, so if local is funder remote doesn't pay the fees
-    val fees = if (commitments1.localParams.isFunder) commitTxFee(commitments1.remoteParams.dustLimit, reduced) else 0.sat
-    val missing = reduced.toRemote - commitments1.remoteParams.channelReserve - fees
-    if (missing < 0.msat) {
-      return Left(InsufficientFunds(commitments.channelId, amount = cmd.amount, missing = -missing.truncateToSatoshi, reserve = commitments1.remoteParams.channelReserve, fees = fees))
+    // note that the funder pays the fee, so if we are fundee the funder needs to afford the increased commitment fee
+    val fees = commitTxFee(commitments1.remoteParams.dustLimit, reduced)
+    val missingForSender = reduced.toRemote.truncateToSatoshi - commitments1.remoteParams.channelReserve - (if (commitments1.localParams.isFunder) fees else 0.sat)
+    val missingForReceiver = reduced.toLocal.truncateToSatoshi - commitments1.localParams.channelReserve - (if (commitments1.localParams.isFunder) 0.sat else fees)
+    if (missingForSender < 0.sat) {
+      return Left(InsufficientFunds(commitments.channelId, amount = cmd.amount, missing = -missingForSender, reserve = commitments1.remoteParams.channelReserve, fees = fees))
+    } else if (!commitments1.localParams.isFunder && missingForReceiver < 0.sat) {
```

Comment:
"recv CMD_SIGN (going above reserve)" test fails without additional `!commitments1.localParams.isFunder` added, as far as I understand this has to do with remote funder's unspendable reserve, but may be mistaken.

Reply:
This error is in fact related to the fundee initially not meeting the `channel_reserve` (e.g. if there was no `push_msat`). In this test bob initially has 0 balance, so even if it doesn't have to pay the commitment tx fee, it is still missing the `channel_reserve`. That's why `missingForReceiver`=`10 000 sat`. I think your change indirectly fixes this but I'd like to find something more straightforward and less ambiguous.

Reply:
Okay, can't find anything better. I'd like to explain more what is going one though:

```scala
    if (missingForSender < 0.sat) {
      return Left(InsufficientFunds(commitments.channelId, amount = cmd.amount, missing = -missingForSender, reserve = commitments1.remoteParams.channelReserve, fees = fees))
    } else if (missingForReceiver < 0.sat) {
      if (commitments.localParams.isFunder) {
        // receiver is fundee; it is ok if it can't maintain its channel_reserve for now, as long as its balance is increasing, which is the case if it is receiving a payment
      } else {
        return Left(RemoteCannotAffordFeesForNewHtlc(commitments.channelId, amount = cmd.amount, missing = -missingForReceiver, reserve = commitments1.remoteParams.channelReserve, fees = fees))
      }
    }
```

Reply:
Ok, I'll use this

Reply:
We should also probably make the check symmetrical and add this to `receiveAdd`:

```scala
// note that the funder pays the fee, so if sender != funder, both sides will have to afford this payment
val fees = commitTxFee(commitments1.remoteParams.dustLimit, reduced)
val missingForSender = reduced.toRemote.truncateToSatoshi - commitments1.localParams.channelReserve - (if (commitments1.localParams.isFunder) 0.sat else fees)
val missingForReceiver = reduced.toLocal.truncateToSatoshi - commitments1.remoteParams.channelReserve - (if (commitments1.localParams.isFunder) fees else 0.sat)
if (missingForSender < 0.sat) {
  throw InsufficientFunds(commitments.channelId, amount = add.amountMsat, missing = -missingForSender, reserve = commitments1.localParams.channelReserve, fees = fees)
} else if (missingForReceiver < 0.sat) {
  if (commitments.localParams.isFunder) {
    throw CannotAffordFees(commitments.channelId, missing = -missingForReceiver, reserve = commitments1.remoteParams.channelReserve, fees = fees)
  } else {
    // receiver is fundee; it is ok if it can't maintain its channel_reserve for now, as long as its balance is increasing, which is the case if it is receiving a payment
  }
}
```